pytest_pyvista/pytest_pyvista.py

```diff
@@ -28,23 +30,90 @@
+
+
+@dataclass
+class _EnvInfo:
+    prefix: str = ""
+    os: bool = True
+    machine: bool = True
+    python: bool = True
+    pyvista: bool = True
+    vtk: bool = True
+    gpu: bool = True
+    ci: bool = True
+    suffix: str = ""
+
+    def __repr__(self) -> str:
+        os_info = _EnvInfo._get_os()
+        os_version = f"{os_info[0]}-{os_info[1]}" if self.os else ""
+        machine = f"{platform.machine()}" if self.machine else ""
+        gpu = f"gpu-{_EnvInfo._gpu_vendor()}" if self.gpu else ""
+        python_version = f"py-{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}" if self.python else ""
+        pyvista_version = f"pyvista-{pyvista.__version__}" if self.pyvista else ""
+        vtk_version = f"vtk-{vtkmodules.__version__}" if self.vtk else ""
+        ci = f"{'' if os.environ.get('CI', None) else 'no-'}CI" if self.ci else ""
+
+        values = [
+            f"{self.prefix}",
+            f"{os_version}",
+            f"{machine}",
+            f"{gpu}",
+            f"{python_version}",
+            f"{pyvista_version}",
+            f"{vtk_version}",
+            f"{ci}",
+            f"{self.suffix}",
         ]
-    )
+        return "_".join(val for val in values if val)
 
+    @staticmethod
+    def _get_os() -> tuple[str, str]:
+        system = platform.system()
+        if system == "Linux":
+            try:
+                name = platform.freedesktop_os_release()["ID"]
+                version = platform.freedesktop_os_release()["VERSION_ID"]
+            except AttributeError:
+                name = system
+                version = platform.release()
+            return name, version
+        name = "macOS" if system == "Darwin" else system
+        return name, platform.release()
 
-ENV_INFO = _get_env_info()
+    @staticmethod
+    def _gpu_vendor() -> str:
+        # Get cached value
+        if _GPU_VENDOR[0]:
+            return _GPU_VENDOR[0]
+
+        try:
+            vendor = pyvista.GPUInfo().vendor
+        except Exception:  # noqa: BLE001  # pragma: no cover
+            vendor = "UNKNOWN"
+
+        # Try to shorten vendor string
+        lower = vendor.lower()
+        if lower.startswith(nv := "nvidia"):  # pragma: no cover
+            text = nv
+        elif lower.startswith(amd := "amd"):  # pragma: no cover
+            text = amd
+        elif lower.startswith(ati := "ati"):  # pragma: no cover
+            text = ati
+        elif lower.startswith(mesa := "mesa"):
+            text = mesa
+        else:
+            text = vendor  # pragma: no cover
+        # Shorten original string and remove whitespace
+        vendor = vendor[: len(text)].replace(" ", "")
+        # Remove all potentially invalid/undesired filename characters
+        disallowed = r'[\\/:*?"<>|\s.\x00]'
+        vendor = re.sub(disallowed, "", vendor)
+
+        # Cache the value globally
+        _GPU_VENDOR[0] = vendor
+        return _GPU_VENDOR[0]
```

Comment:
With `cached_property` suggestion
```suggestion
        return vendor
```